firmware/baseband/proc_btlerx.hpp

```diff
@@ -42,48 +43,196 @@ class BTLERxProcessor : public BasebandProcessor {
+        uint16_t WinOffset;
+        uint16_t Interval;
+        uint16_t Latency;
+        uint16_t Timeout;
+        uint8_t ChM[5];
+        uint8_t Hop;
+        uint8_t SCA;
+    };
+
+    struct ADV_PDU_PAYLOAD_TYPE_R
+    {
+        uint8_t payload_byte[40];
+    };
+
     static constexpr size_t baseband_fs = 4000000;
     static constexpr size_t audio_fs = baseband_fs / 8 / 8 / 2;
 
-    std::array<complex16_t, 512> dst{};
+    uint_fast32_t crc_update(uint_fast32_t crc, const void* data, size_t data_len);
+    uint_fast32_t crc24_byte(uint8_t* byte_in, int num_byte, uint32_t init_hex);
+    bool crc_check(uint8_t* tmp_byte, int body_len, uint32_t crc_init);
+    uint32_t crc_init_reorder(uint32_t crc_init);
+
+    uint32_t crc_initalVale = 0x555555;
+    uint32_t crc_init_internal = 0x00;
+
+    void scramble_byte(uint8_t* byte_in, int num_byte, const uint8_t* scramble_table_byte, uint8_t* byte_out);
+    // void demod_byte(int num_byte, uint8_t *out_byte);
+    int parse_adv_pdu_payload_byte(uint8_t* payload_byte, int num_payload_byte, ADV_PDU_TYPE pdu_type, void* adv_pdu_payload);
+
+    std::array<complex16_t, 1024> dst{};
     const buffer_c16_t dst_buffer{
         dst.data(),
         dst.size()};
 
-    std::array<complex16_t, 512> spectrum{};
-    const buffer_c16_t spectrum_buffer{
-        spectrum.data(),
-        spectrum.size()};
+    static constexpr int RB_SIZE = 2048;
+    uint8_t rb_buf[2048];
 
-    const buffer_s16_t work_audio_buffer{
-        (int16_t*)dst.data(),
-        sizeof(dst) / sizeof(int16_t)};
-
-    // Array size ok down to 375 bauds (24000 / 375)
-    std::array<int32_t, 64> delay_line{0};
-    std::array<int16_t, 1000> rb_buf{0};
-
-    /*dsp::decimate::FIRC8xR16x24FS4Decim8 decim_0 { };
-        dsp::decimate::FIRC16xR16x32Decim8 decim_1 { };
-        dsp::decimate::FIRAndDecimateComplex channel_filter { };*/
     dsp::decimate::FIRC8xR16x24FS4Decim4 decim_0{};
     dsp::decimate::FIRC16xR16x16Decim2 decim_1{};
 
     dsp::demodulate::FM demod{};
     int rb_head{-1};
     int32_t g_threshold{0};
-    uint8_t channel_number{38};
-    int skipSamples{1000};
-    int RB_SIZE{1000};
+    uint8_t channel_number{37};
+
+    uint16_t process = 0;
 
     bool configured{false};
-    AFSKDataMessage data_message{false, 0};
+    BlePacketData blePacketData{};
 
     /* NB: Threads should be the last members in the class definition. */
     BasebandThread baseband_thread{baseband_fs, this, baseband::Direction::Receive};
     RSSIThread rssi_thread{};
 
     void configure(const BTLERxConfigureMessage& message);
+
+    ADV_PDU_PAYLOAD_TYPE_0_2_4_6* payload_type_0_2_4_6 = nullptr;
+    ADV_PDU_PAYLOAD_TYPE_1_3* payload_type_1_3 = nullptr;
+    ADV_PDU_PAYLOAD_TYPE_5* payload_type_5 = nullptr;
+    ADV_PDU_PAYLOAD_TYPE_R* payload_type_R = nullptr;
+    ADV_PDU_PAYLOAD_TYPE_R adv_pdu_payload = {0};
+
+    // clang-format off
+    // Scramble table definition
+    const uint8_t scramble_table[40][42] = 
```

Comment:
hehe much better. Nothing like adding 1000 lines constants to make the file hard to read ;)